pkg/ztunnel/reconciler/table.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconciler
+
+import (
+	"fmt"
+
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/index"
+	"github.com/cilium/statedb/reconciler"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+)
+
+type Namespace struct {
+	*slim_corev1.Namespace
+	// Enrolled indicates if the namespace is enrolled for mTLS.
+	Enrolled                     bool
+	PendingEndpointEnrollment    bool              // PendingEndpointEnrollment indicates if there is a pending endpoint enrollment for the namespace.
+	PendingEndpointDisenrollment bool              // PendingEndpointDisenrollment indicates if there is a pending endpoint disenrollment for the namespace.
+	Status                       reconciler.Status // reconciliation status
+}
```

Comment:
```suggestion
type Namespace struct {
	Name string
	Status                       reconciler.Status // reconciliation status
}
```

If we derive a new MTLSEnrolledNamespace table from the already existing namespace table, I would assume this struct would suffice as a key, wdyt?

Reply:
Yes, I'll keep only the `Enrolled` field

Reply:
The idea is that the table only contains enrolled pods, that should eliminate the need for the Enrolled field.

Reply:
ah gotcha, yea that makes more sense